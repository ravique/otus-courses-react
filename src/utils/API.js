import axios from "axios";

axios.defaults.xsrfCookieName = 'csrftoken';
axios.defaults.xsrfHeaderName = 'X-CSRFToken';

console.log(process.env.REACT_APP_API_URL);

export default axios.create({
  baseURL: process.env.REACT_APP_API_URL,
});