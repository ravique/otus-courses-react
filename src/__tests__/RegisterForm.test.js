import React from 'react';
import {shallowToJson} from 'enzyme-to-json';
import {mount} from "enzyme/build/index";
import '../../jest.config'

import RegisterForm from "../components/pages/register/register_form";
import {MemoryRouter} from "react-router-dom";
import API from "../utils/API";
import MockAdapter from 'axios-mock-adapter';
import {fakeEvent} from "../__mocks__";


describe('RegisterForm', () => {

    let output;
    let mock = new MockAdapter(API);

    beforeEach(() => {
        output = mount(
            <MemoryRouter>
                <RegisterForm/>
            </MemoryRouter>
        ).find('RegisterForm');
    });

    it('should render register form', () => {
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should change state on fields change', () => {
            const mockChangeUsername = {
                target: {
                    name: 'username',
                    value: 'John'
                }
            };

            const mockChangePassword = {
                target: {
                    name: 'password',
                    value: '32768'
                }
            };

            const mockChangePassword2 = {
                target: {
                    name: 'password2',
                    value: '32768'
                }
            };

            const mockChangeEmail = {
                target: {
                    name: 'email',
                    value: 'example@example.com'
                }
            };

            const expected = {
                username: 'John',
                password: '32768',
                email: 'example@example.com'
            };

            output.instance().handleChange(mockChangeUsername);
            output.instance().handleChange(mockChangePassword);
            output.instance().handleChange(mockChangePassword2);
            output.instance().handleChange(mockChangeEmail);

            expect(output.state()).toMatchObject(expected);
        }
    );

    it('should highlight no matching passwords', () => {
        const mockChangePassword = {
            target: {
                name: 'password',
                value: '32768'
            }
        };

        output.setState({
            password2: '1234'
        });

        output.instance().handlePasswordChange(mockChangePassword);

        expect(output.state()).toMatchObject({
            errors: {
                password: ['Passwords mismatch']
            }
        });

        expect(output.text()).toContain('Passwords mismatch');

    });

    it('should highlight no matching passwords', () => {
        const mockChangePassword = {
            target: {
                name: 'password',
                value: '1234'
            }
        };

        output.setState({
            password2: '32768'
        });

        output.instance().handlePasswordChange(mockChangePassword);

        expect(output.state()).toMatchObject({
            errors: {
                password: ['Passwords mismatch']
            }
        });

        expect(output.text()).toContain('Passwords mismatch');

    });

    it('should accept matching passwords', () => {
        const mockChangePassword = {
            target: {
                name: 'password',
                value: '32768'
            }
        };

        output.setState({
            password2: '32768'
        });

        output.instance().handlePasswordChange(mockChangePassword);

        expect(output.state().errors.password).toBeNull();
        expect(output.text()).not.toContain('Passwords mismatch');

    });

    it('should accept register', async () => {
        mock.onPost('register/').reply(200, {"email": "example@example.com"});

        await output.instance().handleSubmit(fakeEvent);

        const success_message = 'Registered successfully. Verification email sent to example@example.com'

        expect(output.state().message).toContain(success_message)
        expect(output.text()).toContain(success_message);

    });

    it('should decline register', async () => {

        mock.onPost('register/').reply(400, {"errors": {"username": ["A user with that username already exists."]}});

        await output.instance().handleSubmit(fakeEvent);

        expect(output.state().errors.username).toContain('A user with that username already exists.');
        expect(output.text()).toContain('A user with that username already exists.');
    })
});