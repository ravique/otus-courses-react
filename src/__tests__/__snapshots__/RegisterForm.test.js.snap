// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render register form 1`] = `
<RegisterForm>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormLabel
      name="username"
    >
      <label
        className="form__label"
        htmlFor="username"
      >
        Username
      </label>
    </FormLabel>
    <FormInput
      eventHandler={[Function]}
      name="username"
      placeholder="Enter your username"
      type="text"
    >
      <input
        className="form__input"
        name="username"
        onChange={[Function]}
        placeholder="Enter your username"
        type="text"
      />
    </FormInput>
    <ThrowErrors />
    <FormLabel
      name="email"
    >
      <label
        className="form__label"
        htmlFor="email"
      >
        Email
      </label>
    </FormLabel>
    <FormInput
      eventHandler={[Function]}
      name="email"
      placeholder="Enter your email"
      type="email"
    >
      <input
        className="form__input"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="email"
      />
    </FormInput>
    <ThrowErrors />
    <FormLabel
      name="password"
    >
      <label
        className="form__label"
        htmlFor="password"
      >
        Password
      </label>
    </FormLabel>
    <FormInput
      eventHandler={[Function]}
      name="password"
      placeholder="Enter your password"
      type="password"
    >
      <input
        className="form__input"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </FormInput>
    <ThrowErrors />
    <FormLabel
      name="password2"
      text="Confirm password"
    >
      <label
        className="form__label"
        htmlFor="password2"
      >
        Confirm password
      </label>
    </FormLabel>
    <FormInput
      eventHandler={[Function]}
      name="password2"
      placeholder="Confirm your password"
      type="password"
    >
      <input
        className="form__input"
        name="password2"
        onChange={[Function]}
        placeholder="Confirm your password"
        type="password"
      />
    </FormInput>
    <ThrowErrors />
    <button
      className="button button_full_width"
      type="submit"
    >
      Register
    </button>
  </form>
</RegisterForm>
`;
